src/index/PatternIndex.cpp

```diff
@@ -0,0 +1,834 @@
+#include "PatternIndex.h"
+#include "MetaDataIterator.h"
+
+const uint32_t PatternIndex::PATTERNS_FILE_VERSION = 1;
+const uint32_t PatternIndex::NAMESPACE_FILE_VERSION = 0;
```

Comment:
Is there a good reason, why these are not either defined and initialized in the header, or completely moved to the implementation file? Const statics (or even static constexpr) can be directly initialized.